File: hello-world-scala/src/main/scala/com/example/scala/MainApp.scala
package com.example.scala

import com.example.scala.helpers.Printer
import com.example.scala.services.{ServiceA, ServiceB, ServiceC}

object MainApp {
  def main(args: Array[String]): Unit = {
    println("hello world Scala")
    Printer.print
    (new ServiceA).print
    (new ServiceB).print
    (new ServiceC).print
  }
}

File: hello-world-java/src/main/java/com/example/java/MainApp.java
package com.example.java;

import com.example.java.helpers.Printer;
import com.example.java.services.ServiceA;
import com.example.java.services.ServiceB;
import com.example.java.services.ServiceC;

class MainApp {
    public static void main(String[] args) {
        System.out.println("hello world Java");
        Printer.print();
        (new ServiceA()).print();
        (new ServiceB()).print();
        (new ServiceC()).print();
    }
}